pkg/reconciler/common/ha.go

```diff
@@ -56,13 +62,17 @@ func HighAvailabilityTransform(instance *servingv1alpha1.KnativeServing, log *za
 				data = map[string]string{}
 			}
 
-			data[enabledComponentsKey] = componentsValue
+			if u.GetNamespace() == "knative-serving" {
+				data[enabledComponentsKey] = servingComponentsValue
+			} else if u.GetNamespace() == "knative-eventing" {
+				data[enabledComponentsKey] = eventingComponentsValue
+			}
```

Comment:
This can't use hardcoded namespaces, imo. We should instead take the KComponent instance and decide based on the type. Or better yet, pass the set of components as a list to the function.

/hold

Reply:
yeah, makes sense - Sorry I forgot to call this out :-)  but it felt already dirty when doing it... But I was heads down w/ my stupid pointer :-)